src/sql/parts/dashboard/tabs/dashboardInnerTab.contribution.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { IExtensionPointUser, ExtensionsRegistry } from 'vs/platform/extensions/common/extensionsRegistry';
+import { IJSONSchema } from 'vs/base/common/jsonSchema';
+import { localize } from 'vs/nls';
+import { join } from 'path';
+import { createCSSRule } from 'vs/base/browser/dom';
+import URI from 'vs/base/common/uri';
+
+import { registerInnerTab, generateInnerTabContentSchemaProperties } from 'sql/platform/dashboard/common/innerTabRegistry';
+
+export type IUserFriendlyIcon = string | { light: string; dark: string; };
+
+export interface IDashboardInnerTabContrib {
+	id: string;
+	title: string;
+	icon?: IUserFriendlyIcon;
+	content: object;
+}
+
+const innerTabSchema: IJSONSchema = {
+	type: 'object',
+	properties: {
+		id: {
+			type: 'string',
+			description: localize('sqlops.extension.contributes.dashboard.innertab.id', "Unique identifier for this inner tab. Will be passed to the extension for any requests.")
+		},
+		icon: {
+			description: localize('sqlops.extension.contributes.dashboard.innertab.icon', '(Optional) Icon which is used to represent this inner tab in the UI. Either a file path or a themable configuration'),
+			anyOf: [{
+				type: 'string'
+			},
+			{
+				type: 'object',
+				properties: {
+					light: {
+						description: localize('carbon.extension.contributes.account.icon.light', 'Icon path when a light theme is used'),
+						type: 'string'
+					},
+					dark: {
+						description: localize('carbon.extension.contributes.account.icon.dark', 'Icon path when a dark theme is used'),
+						type: 'string'
+					}
+				}
+			}]
+		},
+		title: {
+			type: 'string',
+			description: localize('sqlops.extension.contributes.dashboard.innertab.title', "Title of the inner tab to show the user.")
+		},
+		content: {
+			description: localize('sqlops.extension.contributes.dashboard.innertab.content', "The content that will be displayed in this inner tab."),
+			type: 'object',
+			properties: generateInnerTabContentSchemaProperties()
+		}
+	}
+};
+
+const innerTabContributionSchema: IJSONSchema = {
```

Comment:
What if I want to dynamically / conditionally add a tab? Good example is a "Getting Started" guide for a feature that shows only on 1st load, then we save a setting value that indicates it isn't needed. Will we have an ability to define these through Javascript or is the JSON file the only way?

Reply:
Are you referring to alwaysShow property in tab contribution? The current behavior is showing the tab every time. We do have a task to make it behave like pin by default

---
In reply to: [169769876](https://github.com/Microsoft/sqlopsstudio/pull/751#discussion_r169769876) [](ancestors = 169769876)